tests/integration/test_charm.py

```diff
@@ -57,39 +51,33 @@ async def test_s3_storage(
             "s3_access_key_id": localstack_s3_config["credentials"]["access-key"],
             "s3_secret_access_key": localstack_s3_config["credentials"]["secret-key"],
             "s3_server_side_encryption": "false",
-            "s3_tls": "false",
+            "extra_env": '{"MM_FILESETTINGS_AMAZONS3SSL": "false","MM_SERVICESETTINGS_ENABLELOCALMODE": "true","MM_SERVICESETTINGS_LOCALMODESOCKETLOCATION": "/tmp/mattermost.socket"}',
         }
     )
-    await ops_test.model.wait_for_idle(status="active")
+    await model.wait_for_idle(status="active")
 
     # create a user
-    cmd = "MMCTL_LOCAL_SOCKET_PATH=/mattermost/run/local.socket /mattermost/bin/mmctl --local user create --email test@test.test --username test --password thisisabadpassword"
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
+    cmd = f"MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl --local user create --email {test_user['login_id']} --username {test_user['username']} --password {test_user['password']}"
+    await ops_test.juju("run", "--application", app.name, cmd)
 
     # login to the API
-    data = {"login_id": "test@test.test", "password": "thisisabadpassword"}
-    cmd = f"curl -sid '{json.dumps(data)}' http://localhost:8065/api/v4/users/login"
+    cmd = f"curl -sid '{json.dumps(test_user)}' http://localhost:8065/api/v4/users/login"
```

Comment:
Looks like there is still some more `curl` that could be updated to use `requests`?

Reply:
Fixed